src/components/TodoContainer.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import { useState, useEffect } from 'react';
+import AddTodoForm from './AddTodoForm';
+import PropTypes from 'prop-types';
+import TodoList from './TodoList/TodoList';
+
+const TodoContainer = ({ tableName }) => {
+    const [todoList, setTodoList] = useState([]);
+    const [isLoading, setIsLoading] = useState(true);
+
+    const fetchData = async () => {
+        const options = {
+            method: 'GET',
+            headers: {
+                Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`
+            }
+        };
+        const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${tableName}?view=Grid%20view`;
+        try {
+            const response = await fetch(url, options);
+            if (!response.ok) {
+                throw new Error(`Error: ${response.status}`);
+            }
+            const data = await response.json();
+            const todos = data.records.map((todo) => {
+                const newTodo = {
+                    id: todo.id,
+                    title: todo.fields.title
+                };
+                return newTodo;
+            });
+            setTodoList(todos.sort((a, b) => a.title.localeCompare(b.title)));
+            setIsLoading(false);
+        } catch (err) {
+            console.error();
+        }
+    };
+
+    useEffect(() => {
```

Comment:
The useEffect that fetches the data does not include tableName in its dependency array. Since tableName is a prop, if it changes, the fetchData function should be called again to refetch data for the new table.

Also, in React, it's generally considered a good practice to declare your useState variables first, followed by useEffect, and then the rest of the component logic (such as helper functions like fetchData, addTodo, etc.). This helps improve readability and maintain a logical flow in your component.